egui_glium/src/lib.rs

```diff
@@ -185,6 +190,40 @@ pub fn input_to_egui(
                 input_state.raw.scroll_delta += delta;
             }
         }
+        WindowEvent::TouchpadPressure {
+            // device_id,
+            // pressure,
+            // stage,
+            ..
+        } => {
+            // TODO
+        }
+        WindowEvent::Touch(touch) => {
+            let pixels_per_point_recip = 1. / pixels_per_point;
+            let mut hasher = std::collections::hash_map::DefaultHasher::new();
+            touch.device_id.hash(&mut hasher);
+            input_state.raw.events.push(Event::Touch {
+                device_id: TouchDeviceId(hasher.finish()),
+                id: TouchId(touch.id),
+                phase: match touch.phase {
+                    glutin::event::TouchPhase::Started => egui::TouchPhase::Start,
+                    glutin::event::TouchPhase::Moved => egui::TouchPhase::Move,
+                    glutin::event::TouchPhase::Ended => egui::TouchPhase::End,
+                    glutin::event::TouchPhase::Cancelled => egui::TouchPhase::Cancel,
+                },
+                pos: pos2(touch.location.x as f32 * pixels_per_point_recip, 
```

Comment:
```suggestion
                pos: pos2(touch.location.x as f32 * pixels_per_point_recip,
```

Hot tip: configure your editor to run `cargo fmt` on save. It is super nice!

Reply:
Resolved by <https://github.com/emilk/egui/pull/306/commits/b38e2255f347a65d42645ab13ac02223b4aaca1b>.
I actually do have my editor configured to run `cargo fmt` on save, but this particular source file seems to be a bit too much for `cargo fmt`, so it does not format around this peace of code.